### PR TITLE
`azure_cognitive_deployment` - Changes `model.version` property from `Required` to `Optional`.

### PR DESCRIPTION
**Pull request by liuwuliuyun**
`azure_cognitive_deployment` 
- change `model.version` property from `Required` to `Optional`.
- update doc of `model` block for `model.version` property.
- change test case accrodingly.
- Fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/24318

Testing results:
```
GOROOT=C:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"C:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.3\tmp\GoLand\___TestAccCognitiveDeploymentSequential_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_cognitive.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/cognitive #gosetup
"C:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.3\tmp\GoLand\___TestAccCognitiveDeploymentSequential_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_cognitive.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccCognitiveDeploymentSequential\E$ #gosetup
=== RUN   TestAccCognitiveDeploymentSequential
--- PASS: TestAccCognitiveDeploymentSequential (802.33s)
=== RUN   TestAccCognitiveDeploymentSequential/deployment
    --- PASS: TestAccCognitiveDeploymentSequential/deployment (802.33s)
=== RUN   TestAccCognitiveDeploymentSequential/deployment/requiresImport
        --- PASS: TestAccCognitiveDeploymentSequential/deployment/requiresImport (193.04s)
=== RUN   TestAccCognitiveDeploymentSequential/deployment/complete
        --- PASS: TestAccCognitiveDeploymentSequential/deployment/complete (187.35s)
=== RUN   TestAccCognitiveDeploymentSequential/deployment/update
        --- PASS: TestAccCognitiveDeploymentSequential/deployment/update (235.89s)
=== RUN   TestAccCognitiveDeploymentSequential/deployment/basic
        --- PASS: TestAccCognitiveDeploymentSequential/deployment/basic (186.05s)
PASS

Process finished with the exit code 0
```

The `source` property requires a standalone ARM resource id rather than the buildin model and we could not create self trained model for every subscription used for acctests. Therefore I didn't to add `source` to acctests for now. But I think this is a valid property for any user who owns a self trained model and we should include this property in `azurerm_cognitive_deployment`.

### CODE REVIEW

File: internal/services/cognitive/cognitive_deployment_resource.go [hunk 1]
```diff
@@ -98,9 +100,15 @@ func (r CognitiveDeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 						ValidateFunc: validation.StringIsNotEmpty,
 					},
 
+					"source": {
```
Comment (katbyte) [inline internal/services/cognitive/cognitive_deployment_resource.go:103]: as this is a resource id  wouldn't it make more sense to have it be

```suggestion
					"source_resource_id": {
```

File: website/docs/r/cognitive_deployment.html.markdown [hunk 2]
```diff
@@ -66,7 +66,9 @@ A `model` block supports the following:
 
 * `name` - (Required) The name of the Cognitive Services Account Deployment model. Changing this forces a new resource to be created.
 
-* `version` - (Required) The version of Cognitive Services Account Deployment model.
+* `source` - (Optional) Deployment model source ARM resource ID.
```
Comment (katbyte) [inline website/docs/r/cognitive_deployment.html.markdown:69]: what is a valid resource here? could we expand on the docs a bit more

File: internal/services/cognitive/cognitive_deployment_resource_test.go [hunk 3]
```diff
@@ -142,21 +142,21 @@ resource "azurerm_cognitive_account" "test" {
 func (r CognitiveDeploymentTestResource) basic(data acceptance.TestData) string {
```
Comment (katbyte) [inline internal/services/cognitive/cognitive_deployment_resource_test.go:142]: can we add a test for the new property?

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by liuwuliuyun** [comment]
@katbyte , Thanks for the comment. I removed the `model.source` property from this PR since we are not able to test this property for now. The reason is that we dont have a self-trained model in our subscription therefore we dont have a valid ARM resource id for `source` property to test.

**Comment by liuwuliuyun** [comment]
Latest test evidence:

```
GOROOT=C:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"C:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.3\tmp\GoLand\___TestAccCognitiveDeployment_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_cognitive__1_.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/cognitive #gosetup
"C:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.3\tmp\GoLand\___TestAccCognitiveDeployment_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_cognitive__1_.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccCognitiveDeployment_basic\E$ #gosetup
=== RUN   TestAccCognitiveDeployment_basic

--- PASS: TestAccCognitiveDeployment_basic (200.76s)
PASS


Process finished with the exit code 0
```

**Review by stephybun — APPROVED** [review]
Thanks @liuwuliuyun LGTM 👍

**Pull request merged**